src/qat/compiler/transform_passes.py

```diff
@@ -112,31 +112,35 @@ class InputOptimisationResult(ResultInfoMixin):
     optimised_circuit: str = None
 
 
+def get_compiler_config(args):
+    compiler_config = next(
+        (a for a in args if isinstance(a, CompilerConfig)), CompilerConfig()
+    )
```

Comment:
love the default value 👍🏽,

but (this is minor) maybe you want to info-log when no config is found and consequently would be using a default config object -> more verbosity